### PR TITLE
health_check: envoy.reloadable_features.health_check.graceful_goaway_handling deprecation

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Loong <loong.dai@intel.com>

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue] #18438
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: docs/root/version_history/current.rst [hunk 1]
```diff
@@ -25,6 +25,7 @@ Removed Config or Runtime
 * http: removed ``envoy.reloadable_features.require_strict_1xx_and_204_response_headers`` and ``envoy.reloadable_features.send_strict_1xx_and_204_response_headers`` and legacy code paths.
 * http: removed ``envoy.reloadable_features.strip_port_from_connect`` and legacy code paths.
 
+* upstream: removed ``envoy.reloadable_features.health_check.graceful_goaway_handling`` and legacy code paths.
```
Comment (mattklein123) [inline docs/root/version_history/current.rst:28]: Can you merge main and remove the newline above this line? Thank you.

**Review by mattklein123 — CHANGES_REQUESTED** [review]
/wait

File: docs/root/version_history/current.rst (hunk 1 above)
Comment (daixiang0) [reply docs/root/version_history/current.rst:28]: Done.

**Pull request merged**